apps/web/src/components/templates/email-editor/EmailMessagesCards.tsx

```diff
@@ -24,27 +33,157 @@ export function EmailMessagesCards({
+                },
+                {
+                  value: 'mobile',
+                  label: <MobileIcon />,
+                },
+              ]}
+              value={preview}
+              onChange={(value: any) => {
+                setPreview(value);
+              }}
+              defaultValue={preview}
+              fullWidth
+            />
+          </div>
+        </When>
+        <When truthy={view === ViewEnum.EDIT}>
+          <div>
+            <UnstyledButton
+              type="button"
+              onClick={() => {
+                setShowVarManagement(!showVarManagement);
+              }}
+            >
+              {showVarManagement ? <ChevronRight /> : <ChevronLeft />}
+            </UnstyledButton>
+          </div>
+        </When>
+      </Group>
       <When truthy={view === ViewEnum.PREVIEW}>
-        <Preview activeStep={index} />
+        <Preview activeStep={index} view={preview} />
       </When>
       <When truthy={view === ViewEnum.TEST}>
         <TestSendEmail isIntegrationActive={isIntegrationActive} index={index} />
       </When>
       <When truthy={view === ViewEnum.EDIT}>
-        <EmailContentCard
-          key={index}
-          organization={currentOrganization}
-          variables={variables}
-          index={index}
-          isIntegrationActive={isIntegrationActive}
-        />
+        <Grid grow>
+          <Grid.Col span={showVarManagement ? 8 : 12}>
+            <EmailContentCard
+              key={index}
+              organization={currentOrganization}
+              variables={variables}
+              index={index}
+              isIntegrationActive={isIntegrationActive}
+            />
+          </Grid.Col>
+          <Grid.Col
+            span={showVarManagement ? 4 : 0}
+            sx={{
+              maxWidth: '325px',
+            }}
+          >
+            <VariablesManagement
+              index={index}
+              openVariablesModal={() => {
+                setModalOpen(true);
+              }}
+            />
+          </Grid.Col>
+        </Grid>
       </When>
+      <Modal
+        opened={modalOpen}
+        overlayColor={theme.colorScheme === 'dark' ? colors.BGDark : colors.BGLight}
+        overlayOpacity={0.7}
+        styles={{
+          modal: {
+            backgroundColor: theme.colorScheme === 'dark' ? colors.B15 : colors.white,
+            width: '90%',
+          },
+          body: {
+            paddingTop: '5px',
+            paddingInline: '8px',
+          },
+        }}
+        title={<Title>Variables</Title>}
+        sx={{ backdropFilter: 'blur(10px)' }}
+        shadow={theme.colorScheme === 'dark' ? shadows.dark : shadows.medium}
+        radius="md"
+        size="lg"
+        onClose={() => {
+          setModalOpen(false);
+        }}
+        centered
+        overflow="inside"
+      >
+        <VariableManager hideLabel={true} index={index} contents={['content', 'htmlContent', 'subject']} />
```

Comment:
Should this be removed?

Reply:
No this should be here this is required default form